### DISCUSSION TITLE
KeyError: 'global_step' when converting CKPT

### DISCUSSION THREAD

**Discussion by GITwhy18** [azuritecoin/OnnxDiffusersUI#65] [General]
File "c:\SD\convert_original_stable_diffusion_to_diffusers.py", line 799, in <module>
    global_step = checkpoint["global_step"]
KeyError: 'global_step'

when trying to convert existing CKPT files

here the whole out put:
c:\SD\virtualenv\lib\site-packages\numpy\_distributor_init.py:30: UserWarning: loaded more than 1 DLL from .libs:
c:\SD\virtualenv\lib\site-packages\numpy\.libs\libopenblas.FB5AE2TYXYH2IJRDKGDGQ3XBKLKTF43H.gfortran-win_amd64.dll
c:\SD\virtualenv\lib\site-packages\numpy\.libs\libopenblas64__v0.3.21-gcc_10_3_0.dll
  warnings.warn("loaded more than 1 DLL from .libs:"
Traceback (most recent call last):
  File "c:\SD\convert_original_stable_diffusion_to_diffusers.py", line 799, in <module>
    global_step = checkpoint["global_step"]
KeyError: 'global_step'

**Comment by azuritecoin** [comment]
Which model are you trying to convert?

**Reply by GITwhy18** [reply]
any, i have tried Trin, Waifu, ect.

**Reply by azuritecoin** [reply]
You've downloaded the .ckpt file right? For Waifu and Trinart you can use the second conversion script directly to download and convert the model:  
`python convert_stable_diffusion_checkpoint_to_onnx.py --model_path="hakurei/waifu-diffusion" --output_path="model/waifu_diffusion_v1.3_onnx"`  
python convert_stable_diffusion_checkpoint_to_onnx.py --model_path="naclbit/trinart_stable_diffusion_v2" --output_path="model/trinart_sd_v2_onnx"

**Reply by azuritecoin** [reply]
I haven't figured out what the 'global_step' is in some models.

**Reply by caprexy** [reply]
I updated my script and it worked since they added some error handling for global_step
https://github.com/huggingface/diffusers/blob/main/scripts/convert_original_stable_diffusion_to_diffusers.py
